subworkflows/local/bam_joint_calling_germline_sentieon/main.nf

```diff
@@ -0,0 +1,117 @@
+        .groupTuple(by:[0, 3])
+
+    SENTIEON_GVCFTYPER(sentieon_input, fasta, fai, dbsnp, dbsnp_tbi)
+
+    // TO-DO: Try to use a much functionality from Sentieon as possible. Sentieon should have a function for merging vfcs.
+    BCFTOOLS_SORT(SENTIEON_GVCFTYPER.out.vcf_gz)
+
+    gvcf_to_merge = BCFTOOLS_SORT.out.vcf.map{ meta, vcf -> [ meta.subMap('num_intervals') + [ id:'joint_variant_calling', patient:'all_samples', variantcaller:'sentieon_haplotyper' ], vcf ]}.groupTuple()
+
+    // Merge scatter/gather vcfs & index
+    // Rework meta for variantscalled.csv and annotation tools
+    MERGE_GENOTYPEGVCFS(gvcf_to_merge, dict.map{ it -> [ [ id:'dict' ], it ] } )
+
+    vqsr_input = MERGE_GENOTYPEGVCFS.out.vcf.join(MERGE_GENOTYPEGVCFS.out.tbi, failOnDuplicate: true)
+    indels_resource_label = known_indels_vqsr.mix(dbsnp_vqsr).collect()
+    snps_resource_label = known_snps_vqsr.mix(dbsnp_vqsr).collect()
+
+    // Recalibrate INDELs and SNPs separately
+    SENTIEON_VARCAL_INDEL(
+        vqsr_input,
+        resource_indels_vcf,
+        resource_indels_tbi,
+        indels_resource_label,
+        fasta,
+        fai)
+
+    SENTIEON_VARCAL_SNP(
+        vqsr_input,
+        resource_snps_vcf,
+        resource_snps_tbi,
+        snps_resource_label,
+        fasta,
+        fai)
+
+    //Prepare INDELs and SNPs separately for ApplyVQSR
+
+    // Join results of variant recalibration into a single channel tuple
+    // Rework meta for variantscalled.csv and annotation tools
+    vqsr_input_snp = vqsr_input.join(SENTIEON_VARCAL_SNP.out.recal, failOnDuplicate: true)
+        .join(SENTIEON_VARCAL_SNP.out.idx, failOnDuplicate: true)
+        .join(SENTIEON_VARCAL_SNP.out.tranches, failOnDuplicate: true)
+        .map{ meta, vcf, tbi, recal, index, tranche -> [ meta - meta.subMap('id') + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }
+
+    // Join results of variant recalibration into a single channel tuple
+    // Rework meta for variantscalled.csv and annotation tools
+    vqsr_input_indel = vqsr_input.join(SENTIEON_VARCAL_INDEL.out.recal, failOnDuplicate: true)
+        .join(SENTIEON_VARCAL_INDEL.out.idx, failOnDuplicate: true)
+        .join(SENTIEON_VARCAL_INDEL.out.tranches, failOnDuplicate: true)
+        .map{ meta, vcf, tbi, recal, index, tranche -> [ meta - meta.subMap('id') + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }
+
+
+    SENTIEON_APPLYVARCAL_SNP(
+        vqsr_input_snp,
+        fasta,
+        fai)
+
+    SENTIEON_APPLYVARCAL_INDEL(
+        vqsr_input_indel,
+        fasta,
+        fai)
+
+    vqsr_snp_vcf = SENTIEON_APPLYVARCAL_SNP.out.vcf
+    vqsr_indel_vcf = SENTIEON_APPLYVARCAL_INDEL.out.vcf
+
+    //Merge VQSR outputs into final VCF
+    MERGE_VQSR(
+        vqsr_snp_vcf.mix(vqsr_indel_vcf).groupTuple(),
+        dict.map{ it -> [ [ id:'dict' ], it ] }
```

Comment:
you'll need to be careful with the dict channel when syncing with `dev`

Reply:
Yeah, thanks for the warning. I kind of expect some conflicts and bugs when we merge `dev` into the `sentieon`-branch.

I guess we can get some inspiration from `bam_joint_calling_germline_gatk` on how to fix such issues.